### ISSUE TITLE
Make Slick CSP compliant

### ISSUE THREAD

**Issue by ayumihamsaki** [kenwheeler/slick#3799]
I did a quick search and found this issue: https://github.com/kenwheeler/slick/issues/2399

To cut a long story short. All you need to do is remove the injected `style=""` and move the injected CSS into the external CSS file.

Would you be willing to update the code and adding some documentation for people.

**Comment by ahmadalfy** [comment]
That’s interesting I never knew that this violates CSP. The problem is that slick needs to inject inline style because it calculates a lot of things on run-time. Is JSSOR CSP compliant? I would really love to investigate and fix that if it can be done.

**Comment by ayumihamsaki** [comment]
@ahmadalfy yes jssor is fully compliant, have been using their code with a full csp policy for a few years now. All that was needed to get tehir code to pass was to remove the inline styles and replace them with class tags to an external css file.

**Comment by ayumihamsaki** [comment]
> Is JSSOR CSP compliant?

Yes, have been using jssor for a few years with a fully strict csp policy. Three years ago I contacted their support and they gave me instructions how to make their slider fully csp compliant.

You can probably contact them and find out how they done it, because when I spoke to them they gave me more than enough background info on their code.

**Comment by JeanSebTr** [comment]
I'll give a shot at a PR when time allows as our CSP reports are flooded by this, but I though I'd share a few details if someone want to fix this in the meantime.

CSP is not a monolithic thing, but different policies that can be used to restrict what can be done on the page. In that case, the absence of the `unsafe-inline` policy prevent setting the style attribute directly.

As was identified in #2399, it's caused by the reset of the saved style (that reset is also called when initializing slick which is probably when the CSP violations happen for most). I'm not sure what the best fix would be, but the simplest one would be to call `style.clear()` instead of setting the attr when `originalStyling` is empty. This should at least work for the cases where no inline styles were present on the carousel items.

**Comment by JeanSebTr** [comment]
Looks like my brain invented style.clear, this doesn't exist (or more precisely is the clear css prop). This is the equivalent idea:

```
var style = this.style;
for(var i=style.length-1; i>=0; i--) {
    style.removeProperty(style.item(i));
}
```
